meson.build

```diff
@@ -304,6 +304,19 @@ dir_data_extension = dir_data / 'extension'
 
 
 
+###############################################################
+# Library: bonjour
+###############################################################
+
+bonjouropt = get_option('bonjour')
+bonjour = dependency('', required : false)
+if cc.check_header('dns_sd.h', args: g_c_args, required: bonjouropt)
```

Comment:
This isn't quite right either, it checks for the header even if bonjouropt = disabled.

Reply:
I thought same, then checked from https://mesonbuild.com/Build-options.html#features, it says `disabled do not look for the dependency and always return 'not-found'.`

Reply:
```python
        if disabled:
            mlog.log('Check usable header', mlog.bold(hname, True), 'skipped: feature', mlog.bold(feature), 'disabled')
            return False
```

Reply:
Have you checked that that works for check_header()? I.e. that no header lookup is performed when bonjour=disabled?

Reply:
Yes, works for check_header().